src/ComputeSharp.D2D1/Shaders/Interop/Effects/PixelShaderEffect.Properties.cs

```diff
@@ -0,0 +1,259 @@
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        get
+        {
+#if NET6_0_OR_GREATER
+            return &SetConstantBufferImpl;
+#else
+            return (delegate* unmanaged[Stdcall]<IUnknown*, byte*, uint, int>)(void*)Marshal.GetFunctionPointerForDelegate(SetConstantBufferWrapper);
+#endif
+        }
+    }
+
+    /// <inheritdoc cref="D2D1_PROPERTY_BINDING.getFunction"/>
+    [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvStdcall) })]
+    private static int GetConstantBufferImpl(IUnknown* effect, byte* data, uint dataSize, uint* actualSize)
+    {
+        if (data is null)
+        {
+            return E.E_POINTER;
+        }
+
+        PixelShaderEffect* @this = (PixelShaderEffect*)effect;
+
+        if (dataSize < @this->constantBufferSize)
+        {
+            return E.E_NOT_SUFFICIENT_BUFFER;
+        }
+
+        if (@this->constantBuffer is null)
+        {
+            return E.E_NOT_VALID_STATE;
+        }
+
+        uint writtenBytes = 0;
+
+        if (@this->constantBufferSize > 0)
+        {
+            Buffer.MemoryCopy(@this->constantBuffer, data, dataSize, @this->constantBufferSize);
+
+            writtenBytes = (uint)@this->constantBufferSize;
+        }
+
+        if (actualSize is not null)
+        {
+            *actualSize = writtenBytes;
+        }
+
+        return S.S_OK;
+    }
+
+    /// <inheritdoc cref="D2D1_PROPERTY_BINDING.getFunction"/>
+    [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvStdcall) })]
+    private static int SetConstantBufferImpl(IUnknown* effect, byte* data, uint dataSize)
+    {
+        if (data is null)
+        {
+            return E.E_POINTER;
+        }
+
+        PixelShaderEffect* @this = (PixelShaderEffect*)effect;
+
+        if (dataSize != (uint)@this->constantBufferSize)
+        {
+            return E.E_INVALIDARG;
+        }
+
+        if (@this->constantBuffer is not null)
+        {
+            NativeMemory.Free(@this->constantBuffer);
+        }
+
+        void* buffer = NativeMemory.Alloc(dataSize);
+
+        Buffer.MemoryCopy(data, buffer, dataSize, dataSize);
+
+        @this->constantBuffer = (byte*)buffer;
+
+        return S.S_OK;
+    }
+
+    /// <summary>
+    /// Gets the resource texture manager for a given index.
+    /// </summary>
+    /// <param name="index">The index of the resource texture manager to get.</param>
+    /// <param name="data">A pointer to a variable that stores the data that this function retrieves on the property.</param>
+    /// <param name="dataSize">The number of bytes in the property to retrieve.</param>
+    /// <param name="actualSize">A optional pointer to a variable that stores the actual number of bytes retrieved on the property.</param>
+    /// <returns>The <see cref="HRESULT"/> for the operation.</returns>
+    private int GetResourceTextureManagerAtIndex(int index, byte* data, uint dataSize, uint* actualSize)
```

Comment:
should `index` be named `resourceTextureIndex` for clarity?

Reply:
Done in 44bf64741a03d0b66d7ec8d5b2b36acec5050702.